File: Ejercicio2.py
# Verificación de contraseña - Python

STORED_PASSWORD = "Secreto123"  # contraseña almacenada

def verificar_contrasena(entrada, almacenada=STORED_PASSWORD):
    # Comparamos ignorando mayúsculas/minúsculas y espacios extra
    return entrada.strip().lower() == almacenada.strip().lower()

def main():
    intento = input("Escribe la contraseña: ")
    if verificar_contrasena(intento):
        print("Contraseñas coinciden.")
    else:
        print("Contraseñas NO coinciden.")

if __name__ == "__main__":
    main()

File: Ejercicio2.hs
-- Verificación de contraseña - Haskell
import Data.Char (toLower)

-- Función que compara dos Strings ignorando mayúsculas/minúsculas
coincidenIgnoreCase :: String -> String -> Bool
coincidenIgnoreCase a b = map toLower (trim a) == map toLower (trim b)
  where
    -- trim simple para eliminar espacios al inicio/fin
    trim = f . f
      where f = reverse . dropWhile (== ' ')

-- Ejemplo de uso en main (opcional)
main :: IO ()
main = do
  putStrLn "Ingresa la contraseña almacenada:"
  almacenada <- getLine
  putStrLn "Ingresa la contraseña de prueba:"
  entrada <- getLine
  if coincidenIgnoreCase almacenada entrada
    then putStrLn "Contraseñas coinciden."
    else putStrLn "Contraseñas NO coinciden."
